src/main/java/org/truffleruby/core/exception/SyntaxErrorNodes.java

```diff
@@ -35,4 +43,27 @@ RubySyntaxError allocateSyntaxError(RubyClass rubyClass) {
 
     }
 
+    @CoreMethod(names = "path")
+    public abstract static class PathNode extends CoreMethodArrayArgumentsNode {
+
+        @TruffleBoundary
+        @Specialization
+        Object path(RubySyntaxError syntaxError) {
```

Comment:
This is to handle RubyString / Nil - a brief search of the codebase suggested that this was how we handled varying return types, did I miss something?

Reply:
This is correct. Looks great!